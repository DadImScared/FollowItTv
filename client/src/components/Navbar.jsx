
import React from 'react';

import { withStyles } from 'material-ui/styles';
import AppBar from 'material-ui/AppBar';
import Toolbar from 'material-ui/Toolbar';
import Hidden from 'material-ui/Hidden';
import IconButton from 'material-ui/IconButton';

import MenuIcon from 'material-ui-icons/Menu';

import styles from '../styles/Navbar.css';

const Navbar = ({ classes, toggleNav }) => (
  <AppBar color={'secondary'} className={classes.appBar}>
    <Toolbar>
      <Hidden mdUp>
        <IconButton color={'inherit'} onClick={toggleNav}>
          <MenuIcon />
        </IconButton>
      </Hidden>
    </Toolbar>
  </AppBar>
);

export default withStyles(styles)(Navbar);
