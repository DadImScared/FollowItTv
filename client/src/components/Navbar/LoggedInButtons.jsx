
import React from 'react';

import { NavLink, Link } from 'react-router-dom';

import Button from '@material-ui/core/Button';


const LoggedInButtons = () => (
  <div>
    <Button component={NavLink} to={'/my_shows'}>
      My shows
    </Button>
    <Button component={Link} to={'/logout'}>
      Sign out
    </Button>
  </div>
);

export default LoggedInButtons;
