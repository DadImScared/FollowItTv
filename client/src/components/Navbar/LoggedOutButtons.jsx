
import React from 'react';

import { NavLink } from 'react-router-dom';

import Button from '@material-ui/core/Button';


const LoggedOutButtons = () => (
  <div>
    <Button component={NavLink} to={'/login'}>
      Sign in
    </Button>
    <Button component={NavLink} to={'/register'}>
      Sign up
    </Button>
  </div>
);

export default LoggedOutButtons;
