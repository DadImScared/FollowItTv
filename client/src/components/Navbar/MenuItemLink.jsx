
import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import MenuItem from '@material-ui/core/MenuItem';

class MenuItemLink extends Component {

  onClick = () => {
    const { handleClose, routeName } = this.props;
    handleClose(routeName);
  };

  render() {
    return (
      <MenuItem component={Link} to={this.props.routeName} onClick={this.onClick}>
        {this.props.text}
      </MenuItem>
    );
  }
}

MenuItemLink.propTypes = {
  handleClose: PropTypes.func.isRequired,
  routeName: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired
};

export default MenuItemLink;
