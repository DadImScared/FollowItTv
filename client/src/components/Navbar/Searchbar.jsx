
import React, { Component } from 'react';

import TextField from 'material-ui/TextField';
import { InputAdornment } from 'material-ui/Input';
import IconButton from 'material-ui/IconButton';

import Search from 'material-ui-icons/Search';


class Searchbar extends Component {
  constructor(...args) {
    super(...args);
    this.state = {
      inputVal: ''
    };
  }

  handleChange = ({ target: { value: inputVal } }) => {
    this.setState({ inputVal });
  };

  pushSearch = (e) => {
    e.preventDefault();
    const { history: { push } } = this.props;
    const { inputVal } = this.state;
    if (inputVal) {
      push(`/search/${inputVal}`);
      this.setState({ inputVal: '' });
    }
  };

  render() {
    return (
      <form onSubmit={this.pushSearch}>
        <TextField
          InputProps={{
            startAdornment: (
              <InputAdornment position="start">
                <IconButton type={'submit'}>
                  <Search />
                </IconButton>
              </InputAdornment>
            )
          }}
          placeholder={'Search shows'}
          aria-label={'Search shows'}
          value={this.state.inputVal}
          onChange={this.handleChange}
        />
      </form>
    );
  }
}

export default Searchbar;
