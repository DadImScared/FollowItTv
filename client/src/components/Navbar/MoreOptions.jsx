
import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';

import Icon from '@material-ui/core/Icon';
import IconButton from '@material-ui/core/IconButton';
import Menu from '@material-ui/core/Menu';

import MenuItemLink from './MenuItemLink';

export class MoreOptions extends Component {
  constructor(...args) {
    super(...args);
    this.state = {
      anchorEl: null
    };
  }

  handleClick = (event) => {
    this.setState({ anchorEl: event.currentTarget });
  };

  handleClose = () => {
    this.setState({ anchorEl: null });
  };

  render() {
    const { anchorEl } = this.state;
    const { loggedIn } = this.props;
    return (
      <div>
        <IconButton
          aria-owns={anchorEl ? 'simple-menu' : null}
          aria-haspopup="true"
          onClick={this.handleClick}
        >
          <Icon>more_vert</Icon>
        </IconButton>
        <Menu
          anchorEl={anchorEl}
          open={Boolean(anchorEl)}
          onClose={this.handleClose}
        >
          {
            loggedIn ?
              <MenuItemLink handleClose={this.handleClose} routeName={'/logout'} text={'Log out'}/>
              :
              <Fragment>
                <MenuItemLink handleClose={this.handleClose} routeName={'/login'} text={'Sign in'} />
                <MenuItemLink handleClose={this.handleClose} routeName={'/register'} text={'Sign up'} />
              </Fragment>
          }
        </Menu>
      </div>
    );
  }
}

MoreOptions.propTypes = {
  history: PropTypes.shape({
    push: PropTypes.func.isRequired
  }).isRequired,
  loggedIn: PropTypes.bool.isRequired
};

export default MoreOptions;
