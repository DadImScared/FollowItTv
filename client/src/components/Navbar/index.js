
import React from 'react';
import PropTypes from 'prop-types';

import classNames from 'classnames';
import { connect } from 'react-redux';
import { Link, withRouter } from 'react-router-dom';

import { withStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Hidden from '@material-ui/core/Hidden';
import Button from '@material-ui/core/Button';

import LoggedInButtons from './LoggedInButtons';
import LoggedOutButtons from './LoggedOutButtons';
import MoreOptions from './MoreOptions';
import Searchbar from './Searchbar';
import styles from '../../styles/Navbar.css';

export const Navbar = ({ classes, loggedIn, directionDown, ...other }) => {
  return (
    <AppBar
      color={'secondary'}
      className={classNames(classes.appBar, {
        [classes.hideNavBar]: directionDown
      })}
    >
      <Toolbar style={{ justifyContent: 'space-between' }}>
        <Hidden smUp>
          <div />
        </Hidden>
        <div>
          <Searchbar {...other} />
        </div>
        <Hidden smDown>
          <div style={{ display: 'flex' }}>
            <Button component={Link} to={'/schedule'}>
              Schedule
            </Button>
            {
              loggedIn ?
                <LoggedInButtons />
                :
                <LoggedOutButtons/>
            }
          </div>
        </Hidden>
        <Hidden mdUp>
          <div>
            <MoreOptions {...other} loggedIn={loggedIn} />
          </div>
        </Hidden>
      </Toolbar>
    </AppBar>
  );
};

Navbar.propTypes = {
  classes: PropTypes.shape({
    appBar: PropTypes.string.isRequired,
    hideNavBar: PropTypes.string.isRequired
  }).isRequired,
  loggedIn: PropTypes.bool.isRequired,
  directionDown: PropTypes.bool.isRequired
};

const mapStateToProps = ({ users: { loggedIn } }) => ({
  loggedIn
});

export default withStyles(styles)(withRouter(connect(mapStateToProps)(Navbar)));
