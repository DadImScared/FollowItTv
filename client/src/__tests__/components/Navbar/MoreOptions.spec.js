
import React from 'react';

import sinon from 'sinon';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { MoreOptions } from '../../../components/Navbar/MoreOptions';

describe('MoreOptions', () => {
  it('should render logged in options if loggedIn', () => {
    const wrapper = shallow(<MoreOptions history={{ push: sinon.spy() }} loggedIn={true} logOut={sinon.spy()}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render logged out options if loggedIn false', () => {
    const wrapper = shallow(<MoreOptions history={{ push: sinon.spy() }} loggedIn={false} logOut={sinon.spy() }/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
