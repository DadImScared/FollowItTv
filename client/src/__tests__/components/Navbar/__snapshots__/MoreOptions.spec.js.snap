// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoreOptions should render logged in options if loggedIn 1`] = `
<div>
  <WithStyles(IconButton)
    aria-haspopup="true"
    aria-owns={null}
    onClick={[Function]}
  >
    <WithStyles(Icon)>
      more_vert
    </WithStyles(Icon)>
  </WithStyles(IconButton)>
  <WithStyles(Menu)
    anchorEl={null}
    onClose={[Function]}
    open={false}
  >
    <WithStyles(MenuItem)
      onClick={[Function]}
    >
      Log out
    </WithStyles(MenuItem)>
  </WithStyles(Menu)>
</div>
`;

exports[`MoreOptions should render logged out options if loggedIn false 1`] = `
<div>
  <WithStyles(IconButton)
    aria-haspopup="true"
    aria-owns={null}
    onClick={[Function]}
  >
    <WithStyles(Icon)>
      more_vert
    </WithStyles(Icon)>
  </WithStyles(IconButton)>
  <WithStyles(Menu)
    anchorEl={null}
    onClose={[Function]}
    open={false}
  >
    <React.Fragment>
      <MenuItemLink
        handleClose={[Function]}
        routeName="/login"
        text="Sign in"
      />
      <MenuItemLink
        handleClose={[Function]}
        routeName="/register"
        text="Sign up"
      />
    </React.Fragment>
  </WithStyles(Menu)>
</div>
`;
